src/WordPress/PluginActions.php

```diff
@@ -6,139 +6,39 @@
 use CF\API\Request;
 use CF\API\Plugin;
 use CF\Integration\DefaultIntegration;
-use CF\WordPress\Constants\Exceptions\PageRuleLimitException;
-use CF\WordPress\Constants\Exceptions\ZoneSettingFailException;
+use CF\API\Exception\PageRuleLimitException;
+use CF\API\Exception\ZoneSettingFailException;
 use CF\WordPress\Constants\Plans;
+use CF\API\AbstractPluginActions;
 
-class PluginActions
+class PluginActions extends AbstractPluginActions
 {
-    private $api;
-    private $config;
-    private $wordpressAPI;
-    private $dataStore;
-    private $logger;
-    private $request;
+    private $defaultIntegration;
     private $wordPressClientAPI;
 
-    /**
-     * @param DefaultIntegration $defaultIntegration
-     * @param APIInterface       $api
-     * @param Request            $request
-     */
     public function __construct(DefaultIntegration $defaultIntegration, APIInterface $api, Request $request)
     {
-        $this->api = $api;
-        $this->config = $defaultIntegration->getConfig();
-        $this->wordpressAPI = $defaultIntegration->getIntegrationAPI();
-        $this->dataStore = $defaultIntegration->getDataStore();
-        $this->logger = $defaultIntegration->getLogger();
-        $this->request = $request;
-
-        $this->wordPressClientAPI = new WordPressClientAPI($defaultIntegration);
+        parent::__construct($defaultIntegration, $api, $request);
+        $this->defaultIntegration = $defaultIntegration;
     }
 
-    /**
-     * GET /zones.
-     *
-     * @return mixed
-     */
-    public function loginWordPress()
+    public function createWordPressClientAPI(DefaultIntegration $defaultIntegration)
     {
-        $apiKey = $this->request->getBody()['apiKey'];
-        $email = $this->request->getBody()['email'];
-
-        $isCreated = $this->dataStore->createUserDataStore($apiKey, $email, null, null);
-
-        if (!$isCreated) {
-            $this->logger->error('Creating user data to store failed');
-
-            return $this->api->createAPIError('Unable to save user credentials');
-        }
-
-        $response = $this->api->createAPISuccessResponse(array('email' => $email));
-
-        return $response;
+        return new WordPressClientAPI($defaultIntegration);
     }
 
-    /**
-     * GET /plugin/:zonedId/settings.
+    /*
+     * PATCH /plugin/:id/settings/default_settings               
      *
-     * @return mixed
+     * Requests are syncronized
      */
-    public function getPluginSettings()
+    public function applyDefaultSettings()
     {
-        $settings = $this->dataStore->getPluginSettings();
+        $this->wordPressClientAPI = $this->createWordPressClientAPI($this->defaultIntegration);
```

Comment:
Why wrap this in a method since `createWordPressClientAPI()` just returns it with no extra logic?